packages/x-tree-view/src/internals/plugins/useTreeViewExpansion/useTreeViewExpansion.ts

```diff
@@ -1,50 +1,56 @@
 import * as React from 'react';
 import useEventCallback from '@mui/utils/useEventCallback';
 import { TreeViewPlugin } from '../../models';
-import { populateInstance } from '../../useTreeView/useTreeView.utils';
+import { populateInstance, populatePublicAPI } from '../../useTreeView/useTreeView.utils';
 import { UseTreeViewExpansionSignature } from './useTreeViewExpansion.types';
+import { TreeViewItemId } from '../../../models';
 
 export const useTreeViewExpansion: TreeViewPlugin<UseTreeViewExpansionSignature> = ({
   instance,
+  publicAPI,
   params,
   models,
 }) => {
+  const expandedItemsMap = React.useMemo(() => {
```

Comment:
I took the opportunity to do this performance improvement now because in order to avoid code duplication between `toggleItemExpansion` and `setItemExpansion`, I need to check the expansion status twice and doing it with a linear complexity with awefull.

@romgrk do you have a preference on the data structure to use for this model? Knowing that for now it's purely internal and we keep the array for the `expandedItems` prop (we can reconsider in the future).